### PR TITLE
Feat/corpcal 287 configurable info icons

### PR DESCRIPTION
**Pull request by AlexCarmichael-bcgov**
can now set which activity fields have info icons, and what information is displayed.

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
     "lib": ["ES2022"],
     "module": "ESNext",
     "moduleResolution": "bundler",
+    "ignoreDeprecations": "5.0",
```
Comment (cgovbc) [inline tsconfig.base.json:7]: Seems unrelated? Maybe we can move into a separate PR if we need this.

**Review by cgovbc — APPROVED** [review]
LGTM! Nice work!
Nits from the robot below. IMO only the first 3 suggestions are important. I can give more clarity on what "Medium — Composite labels can show multiple info icons" means, if you need.

Findings below are sorted by severity.

---

### High — Admin can save while server settings never loaded successfully

**File:** `calendar-ui/src/components/admin/ActivityInfoIconSettingsAdmin.tsx`

The admin query always supplies `initialData` (cache or `DEFAULT_ACTIVITY_INFO_ICON_SETTINGS`). On fetch failure, `data` is still populated, the editor renders (with amber banners), and **Save stays enabled** when `hasChanges` and validation pass.

**Impact:** During an outage or transient API failure, a system admin can PATCH **defaults** or **stale localStorage** over production settings without a successful server read.

**Recommended fix:** Disable Save unless the query succeeded (`isSuccess` / `!isError`), or use `placeholderData` instead of `initialData` for display-only fallback. At minimum, disable Save when `isUsingDefaultFallback` is true; consider the same when `error` is set unless product explicitly allows “save cache to server.”

---

### Medium — Composite labels can show multiple info icons

**Files:** `ActivityScheduleSection.tsx`, `ActivityEventSection.tsx`

Grouped labels (“Date”, “Time”, “Venue”) mount separate `ActivityFieldInfoIcon` instances per underlying field. Changed badges already use local `DATE_GROUP_FIELDS` / `TIME_GROUP_FIELDS` and `FormAggregateDirtyIndicator`; info icons do not share that pattern.

**Impact:** If an admin enables icons for several sub-fields, users see multiple icons on one label—confusing and inconsistent with the single aggregate “Changed” badge.

**Recommended fix:** Introduce shared `ACTIVITY_FORM_FIELD_GROUPS` in `packages/shared` (paths for badges, `infoIconKeys` for settings) and a single `ActivityFormGroupInfoIcon` with a documented text rule (first configured key or merged plain text). Refactor schedule/venue labels to use it.

---

### Medium — Rich-text change detection uses raw strings

**File:** `calendar-ui/src/components/admin/ActivityInfoIconSettingsAdmin.tsx` (`hasChanges` vs `invalidFieldKeys` / persisted copy)

`invalidFieldKeys` and “Saved / Unsaved changes” use `plainTextFromActivityRichField`, but `hasChanges` compares **raw stored values** with `.trim()`.

**Impact:** After WYSIWYG, the same visible text can differ in JSON vs markdown storage, so Save or “Unsaved changes” can be wrong.

**Recommended fix:** Use one normalized comparator everywhere (`plainTextFromActivityRichField` and/or canonical stored JSON via existing coalesce helpers) in `hasChanges`, validation, persisted-state UI, and optionally the PATCH payload.

---

### Medium — No automated tests for backend settings API

**Files:** `calendar-service/src/locks/activity-info-icons.controller.ts`, `application-settings.service.ts` (`getActivityInfoIconSettings` / `setActivityInfoIconSettings`)

There are still no `calendar-service` specs for GET/PATCH or service read/write behavior.

**Impact:** Regressions in auth, validation, persistence, or empty-config handling may ship without CI signal.

**Recommended fix:** Add controller/service tests mirroring `review-exempt-fields` patterns: happy path, invalid JSON fallback, empty `items`, permission on PATCH, open GET for authenticated users (if that remains intentional).

---

### Low — PATCH body is not normalized before send

**File:** `calendar-ui/src/components/admin/ActivityInfoIconSettingsAdmin.tsx` (save payload)

Save sends `(draft.textsByFieldKey[fieldKey] ?? '').trim()` while the server validates plain-text length via the shared schema.

**Impact:** Edge cases (empty TipTap doc JSON, odd whitespace) can cause avoidable 400s or inconsistent stored shapes vs other rich-text fields.

**Recommended fix:** Normalize each item through the same path used elsewhere (`coalesceRichTextFormStorageValue` or client-side `activityInfoIconSettingsSchema.parse`) before PATCH.

---

### Low — No automated coverage for “clear all icons” contract

**Files:** `activity-info-icon-settings.schema.spec.ts`, `application-settings.service.ts`, optional `activity-info-icon-settings-context.test.tsx`

Empty `items` is allowed at read time after the service fix, but there is no test for PATCH/GET `[]` or form behavior with no configured icons.

**Impact:** The old “empty save falls back to defaults” bug could return unnoticed.

**Recommended fix:** Schema test for `{ items: [] }`; service test for set/get roundtrip; optional context test that no icon renders when the map is empty.

---

### Low — Save allowed when refresh failed on cached fallback

**File:** `calendar-ui/src/components/admin/ActivityInfoIconSettingsAdmin.tsx` (`isUsingCachedFallback`)

User-scoped cache reduces cross-user risk, but Save stays enabled on query error when cached data is shown.

**Impact:** An admin can overwrite newer server settings with an older cache if another admin changed settings after this client last fetched.

**Recommended fix:** Disable Save when `error` is set, or require confirmation before save on cached fallback; align with the High finding above.

---

### Low — Admin hard error branch is effectively dead

**File:** `calendar-ui/src/components/admin/ActivityInfoIconSettingsAdmin.tsx` (`error ? "Could not load settings"`)

With `initialData` always set, `!isLoading && data` wins on error; users see banners plus an editable draft instead of the hard error message.

**Impact:** Misleading code path and no distinct UX when load truly failed without a trustworthy draft.

**Recommended fix:** Gate the editor on `isSuccess` (or explicit fallback rules); remove or rewrite the dead branch to match banner-only behavior.

File: tsconfig.base.json (hunk 1 above)
Comment (AlexCarmichael-bcgov) [reply tsconfig.base.json:7]: yeah, let's nix this change.
